catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/MgrsCoordinateProcessor.java

```diff
@@ -0,0 +1,354 @@
+        LOGGER.trace(
+            "Ignoring this part, since NUMERIC is only valid in a ZONE but no ZONE was provided");
+        break;
+    }
+  }
+
+  /**
+   * Given an MGRS {@link MgrsPartType#NUMERIC} part format the string to ensure an accurate
+   * conversion, using whitespace as a hint as to the user's intentions.
+   *
+   * <p>The usng4j library demands a minimum of 6 symbols if the easting / northing data is not to
+   * be ignored. That is, the easting component must be fully specified for the conversion to work.
+   * To accomodate this, zeros are prepended onto the easting to whatever appropriate degree.
+   *
+   * <p>For example, an input of <span>4Q KJ 1 2</span> will become <span>4QKJ000012</span> so the
+   * easting / northing is not ignored.
+   *
+   * <p>Refer to the unit tests for more comprehensive examples.
+   *
+   * <p>TODO DDF-4243 - This should be simplified if usng4j is improved
+   *
+   * @see #assemble(String, String)
+   * @param mgrsPart an MGRS substring that contains BOTH the easting and northing.
+   * @return a modified easting/northing string with zeros backfilled to ensure accurate conversion.
+   */
+  @SuppressWarnings("squid:S1135" /* Ticket number provided on to-do comment */)
+  private static String processNumericType(String mgrsPart) {
+    String[] numericParts = mgrsPart.split("[ ]+");
+    if (numericParts.length >= 3 || numericParts.length == 0) {
+      throw new IllegalArgumentException(
+          "The mgrs part provided is not a valid NUMERIC type: " + mgrsPart);
+    }
+    if (numericParts.length == 2) {
+      return assemble(numericParts[0], numericParts[1]);
+    }
+    String eastingAndNorthing = numericParts[0];
+    if (eastingAndNorthing.length() % 2 > 0) {
+      eastingAndNorthing = eastingAndNorthing.substring(0, eastingAndNorthing.length() - 1);
+    }
+    int halfway = eastingAndNorthing.length() / 2;
+    String easting = eastingAndNorthing.substring(0, halfway);
+    String northing = eastingAndNorthing.substring(halfway, eastingAndNorthing.length());
+    return assemble(easting, northing);
+  }
+
+  private static String assemble(String easting, String northing) {
+    StringBuilder numericBuilder = new StringBuilder();
+
+    if (easting.length() < MAX_EASTING_LENGTH) {
+      int numberOfZerosNeeded = MAX_EASTING_LENGTH - easting.length();
+      numericBuilder.append(StringUtils.repeat(ZERO, numberOfZerosNeeded));
+    }
+    numericBuilder.append(easting);
+
+    if (northing.length() < MAX_NORTHING_LENGTH) {
+      int numberOfZerosNeeded = MAX_NORTHING_LENGTH - northing.length();
+      numericBuilder.append(StringUtils.repeat(ZERO, numberOfZerosNeeded));
+    }
+    numericBuilder.append(northing);
+
+    return numericBuilder.toString();
+  }
+
+  private static MgrsPartType getPartType(String mgrsPart) {
+    if (PATTERN_MGRS_ZONE.matcher(mgrsPart).matches()) {
+      return MgrsPartType.ZONE;
+    } else if (PATTERN_MGRS_100KM.matcher(mgrsPart).matches()) {
```

Comment:
:pencil2: The `else` clauses seem redundant, they would be needed if a single return of a mutated variable was used.

Reply:
Not sure what else I could do here. I only want to map to a type if the regex is a match, and in this priority order.

Reply:
Unless I am missing something, the following should work:
```java
if (PATTERN_MGRS_ZONE.matcher(mgrsPart).matches()) return MgrsPartType.ZONE;
if (PATTERN_MGRS_100KM.matcher(mgrsPart).matches()) return MgrsPartType.HUNDRED_KM;
if (PATTERN_MGRS_NUMERIC.matcher(mgrsPart).matches()) return MgrsPartType.NUMERIC;
return MgrsPartType.NONE;
```